### ISSUE TITLE
Duo push without redirect bug

### ISSUE THREAD

**Issue by nightah** [clems4ever/authelia#345]
If you attempt to visit your auth portal directly and have Duo selected as the second factor method after getting a push request and accepting on your said device the result always appears to be failed per the following screenshot:

![duofailed](https://upload.nerv.com.au/selif/chrome2019-03-2722-07-20.png)

The logs have the following detail:
```
debug: date='Wed Mar 27 2019 11:06:51 GMT+0000 (UTC)' method='POST', path='/api/duo-push' requestId='e9d2bcab-132e-4832-9a21-157ff4b14085' sessionId='az2Bj1mnigGjgebjrbThIHVbgSA-lgMB' ip='10.10.10.254' message='Access allowed by user via Duo Push.'
debug: date='Wed Mar 27 2019 11:06:51 GMT+0000 (UTC)' method='POST', path='/api/duo-push' requestId='e9d2bcab-132e-4832-9a21-157ff4b14085' sessionId='az2Bj1mnigGjgebjrbThIHVbgSA-lgMB' ip='10.10.10.254' message='Header x-target-url is set to undefined'
```
If you refresh the page it will show that you are authenticated.

If attempting the same test with TOTP the following occurs in the log:
```
info: date='Wed Mar 27 2019 11:10:54 GMT+0000 (UTC)' method='POST', path='/api/totp' requestId='a7a3a091-c0d9-4534-aa4c-c182612de053' sessionId='az2Bj1mnigGjgebjrbThIHVbgSA-lgMB' ip='10.10.10.254' message='Initiate TOTP validation for user "Amir".'
debug: date='Wed Mar 27 2019 11:10:54 GMT+0000 (UTC)' method='POST', path='/api/totp' requestId='a7a3a091-c0d9-4534-aa4c-c182612de053' sessionId='az2Bj1mnigGjgebjrbThIHVbgSA-lgMB' ip='10.10.10.254' message='TOTP validation succeeded.'
debug: date='Wed Mar 27 2019 11:10:54 GMT+0000 (UTC)' method='POST', path='/api/totp' requestId='a7a3a091-c0d9-4534-aa4c-c182612de053' sessionId='az2Bj1mnigGjgebjrbThIHVbgSA-lgMB' ip='10.10.10.254' message='Header x-target-url is set to undefined'
```

The TOTP request also results in a tick and the user being classified as Authenticated without refreshing the page.

**Comment by clems4ever** [comment]
Hello @nightah , I think I'm missing something in the test case. Can you please clarify the following?
- Do you see the red tick on first validation of the Duo factor or after several attempts?
- Do you have a default redirection URL set in the configuration?
- Do you have an error message in the browser console?

I don't reproduce the error in the duo-push suite. In this suite when I validate the Duo factor, I get redirected to the default URL which is home.example.com.
So basically, if you access the portal without redirection URL, I expect the user to be redirected to the default URL if configured or see the "Already authenticated" message if default URL is not provided.

**Comment by nightah** [comment]
Fixed by #346.
